rs/http_endpoints/public/src/nns_delegation_manager.rs

```diff
@@ -0,0 +1,407 @@
+                _server_name: &ServerName,
+                _ocsp_response: &[u8],
+                _now: UnixTime,
+            ) -> Result<ServerCertVerified, rustls::Error> {
+                Ok(ServerCertVerified::assertion())
+            }
+            fn verify_tls12_signature(
+                &self,
+                _: &[u8],
+                _: &CertificateDer<'_>,
+                _: &DigitallySignedStruct,
+            ) -> Result<HandshakeSignatureValid, rustls::Error> {
+                Ok(HandshakeSignatureValid::assertion())
+            }
+            fn verify_tls13_signature(
+                &self,
+                _: &[u8],
+                _: &CertificateDer<'_>,
+                _: &DigitallySignedStruct,
+            ) -> Result<HandshakeSignatureValid, rustls::Error> {
+                Ok(HandshakeSignatureValid::assertion())
+            }
+            fn supported_verify_schemes(&self) -> Vec<SignatureScheme> {
+                rustls::crypto::ring::default_provider()
+                    .signature_verification_algorithms
+                    .supported_schemes()
+            }
+        }
+
+        let accept_any_config = ClientConfig::builder()
+            .dangerous()
+            .with_custom_certificate_verifier(Arc::new(NoVerify))
+            .with_no_client_auth();
+
+        let mut tls_config = MockTlsConfig::new();
+        tls_config
+            .expect_client_config()
+            .returning(move |_, _| Ok(accept_any_config.clone()));
+
+        (registry_client, tls_config)
+    }
+
+    #[test]
+    fn nns_test() {
+        let rt = tokio::runtime::Runtime::new().unwrap();
+        let (registry_client, tls_config) = set_up(rt.handle().clone());
+
+        let (_, rx) = start_nns_delegation_manager(
+            &MetricsRegistry::new(),
+            Config::default(),
+            no_op_logger(),
+            rt.handle().clone(),
+            NNS_SUBNET_ID,
+            NNS_SUBNET_ID,
+            registry_client,
+            Arc::new(tls_config),
+            CancellationToken::new(),
+        );
+
+        assert!(rx.borrow().is_none());
+    }
+
+    #[test]
+    fn non_nns_test() {
+        let rt = tokio::runtime::Runtime::new().unwrap();
+        let (registry_client, tls_config) = set_up(rt.handle().clone());
+
+        let (_, rx) = start_nns_delegation_manager(
+            &MetricsRegistry::new(),
+            Config::default(),
+            no_op_logger(),
+            rt.handle().clone(),
+            NON_NNS_SUBNET_ID,
+            NNS_SUBNET_ID,
+            registry_client,
+            Arc::new(tls_config),
+            CancellationToken::new(),
+        );
+
+        let delegation = rx
+            .borrow()
+            .clone()
+            .expect("Should return Some delegation on non NNS subnet");
+        let parsed_delegation: Certificate = serde_cbor::from_slice(&delegation.certificate)
+            .expect("Should have returned a valid certificate");
```

Comment:
IIUC we technically don't know whether it is valid at this point, we only know that it deserializes correctly, no? We should probably also validate it here.